File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/entity/EventBooking.ts
import {TypeEventBookingStatus} from "./TypeEventBookingStatus";
import {UserModel} from "./User.model";

export class EventBooking {
  bookingEventId!: number;
  numberOfTickets!: number;
  totalAmount!: number;
  pricePerTicket!: number;
  event!: Event;

  cardNumber!: String;
  cardHolderName!: String;
  expirationDate!: String;
  cvv!: String;
  user!: UserModel;
}

File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/entity/EventBooking.java
package tn.esprit.coexist.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class EventBooking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer bookingEventId;
    int numberOfTickets;
    double totalAmount;


    String cardNumber;
    String cardHolderName;
    String expirationDate;
    String cvv;
    @ManyToOne
    private Event event;

    @ManyToOne
    private User user;




}